examples/step-65/doc/intro.dox

```diff
@@ -0,0 +1,389 @@
+for details.
+
+To give an example, consider the following situation of a two-dimensional
+annulus (with pictures taken from the manifold module). If we start with an
+initial mesh of 10 cells and refine the mesh three times globally without
+attaching any manifolds, we would obtain the following mesh:
+
+@image html hypershell-nothing.png ""
+
+Obviously, we must attach a curved description to the boundary faces of the
+triangulation to reproduce the circular shape upon mesh refinement, like in
+the following picture
+
+@image html hypershell-boundary-only.png ""
+
+However, the mesh in this picture is still not optimal for an annulus in the
+sense that the lines from one cell to the next have kinks at certain vertices,
+and one would rather like to use the following mesh:
+
+@image html hypershell-all.png ""
+
+In this last (optimal) case, which is also the default produced by
+GridGenerator::hyper_shell(), the curved manifold description (in this case a
+polar manifold description) is applied not only to the boundary faces, but to
+the whole domain. Whenever the triangulation requests a new point, e.g. the
+mid point of the edges or the cells when it refines a cell into four children,
+it will place them along the respective mid points in the polar coordinate
+system. By contrast, the case above where only the boundary was subject to the
+polar manifold, only mid points along the boundary would be placed along the
+curved description, whereas mid points in the interior would be computed by
+suitable averages of the surrounding points in the Cartesian coordinate system
+(see the @ref manifold "manifold module" for more details).
+
+At this point, one might assume that curved volume descriptions are the way to
+go. However, this becomes impossible for as simple a case as the
+two-dimensional disk because the polar manifold degenerates in the origin and
+would not produce reasonable new points. A similar thing happens at the origin
+of the three-dimensional ball when one tries to attach a spherical manifold to
+the whole volume &endash; in this case, the computation of new manifold points
+would abort with an exception. These two simple examples make it clear that
+for many interesting cases we must step back from the desire to have an
+analytic curved description for the full volume. This is particularly true if
+the boundary description is provided by some CAD files like in the step-54
+tutorial program, which are intrinsically surface-only.
+
+Yet, a curved boundary description alone is sometimes not enough. Consider the
+case of a torus (e.g. generated with GridGenerator::torus()) with a
+TorusManifold object attached to the surface only, no additional manifolds on
+the interior cells and faces, and with six cells in toroidal direction before
+refinmennt. If the mesh is refined once, we would obtain the following mesh,
+shown with the upper half of the mesh clipped away:
+
+@image html torus_no_inner_manifold.png ""
```

Comment:
Yes, we put most tutorial images on the website. I was a bit split for this tutorial, though. I do use a few pictures we have in the github repository already (three hypershell pictures above, and two of the three "circular_mesh..." pictures below), because they are used in the manifold module. Therefore, I thought it makes most sense to put the four additional generic pictures that are related to explanations rather than the tutorial output itself, `circular_mesh_boundary_cells.png`, `torus_no_inner_manifold.png`, `torus_cylindrical_inner_manifold.png`, `torus_transfinite_manifold.png`, also into the `doc/doxygen/images` subdirectories. But I am definitely open to change that if we have a consensus of how to deal with this issue. Or more generally, how much binary data we think is better handled on the homepage than within the git repository

Reply:
Our workflow of uploading/linking external files is a bit annoying (and we should fix this at some point, for example using a separate git repo everybody can commit to), but I don't see a great advantage of shipping any images to the user in the main repo.
Anyways, not a big deal and it looks like you spent some thought here (generic vs step images).